activerecord/test/schema/postgresql_specific_schema.rb

```diff
@@ -46,6 +47,31 @@
 "
   end
 
+  begin
+    execute <<_SQL
+    CREATE OR REPLACE FUNCTION calculate_defaults()
+    RETURNS TRIGGER AS $$
+    BEGIN
+      NEW.random_number_plus_two := NEW.random_number + 2;
+      RETURN NEW;
+    END;
+    $$
+    LANGUAGE plpgsql;
+
+    CREATE TRIGGER insert_or_update_defaults_trigger
+      BEFORE INSERT OR UPDATE ON defaults
+      FOR EACH ROW EXECUTE PROCEDURE calculate_defaults();
+_SQL
+  rescue ActiveRecord::StatementInvalid => e
+    if e.message.include?('language "plpgsql" does not exist')
+      execute "CREATE LANGUAGE 'plpgsql';"
```

Comment:
Any reason to not make this into schema at the beginning of PG test setup? This make all tests running before (random set) being in environment without this, and all tests (the rest of the random set) with this.

Reply:
@simi 

It's probably a good idea to keep the database configuration concerns together at the top of this file. However, I am confused what you mean by some tests running without this? Could you provide instructions I can execute to see what you mean?

Correct me if I'm wrong, but all the statements defined here get executed only once before the test suite runs? That is to say 'plpgsql' will exist in the environment for every test that runs with the PostgreSQL adapter defined.

If I comment out the silencer here:

https://github.com/rails/rails/blob/e73837a5e680c7eeeef2303f043e809fa478245f/activerecord/test/support/load_schema_helper.rb#L6-L7

and run the suite, the console output seems to support that?

Reply:
@abaldwin88 ahh, you're right. Sorry for confusion. Also I see this pattern is already used in this file. Even I'm not sure it is best practice to let DB fail hard, rescue and retry, your implementation just copies patterns used around. :+1:

Reply:
@simi No worries - thanks for reviewing!